Update test cases for schedule and unschedule command (#326)
AY2324S2-CS2103-F15-1/tp: docs/DeveloperGuide.md

- `remark 100000 r/Birthday on 31 July.`.
- `remark 1 r/Birthday on 31 July r/`.

### Scheduling a meeting with a person

#### Scheduling a meeting with a person successfully

> 

Reviewer: Would adding this line be better:
> Note that The DateTime values specified for this example **must be after system time**.

**Prerequisites:**
1. List all persons using the `list` command. At least 5 persons are in the list.
2. Assume the time for now is 14-04-2024T12:00.
3. Assume no scheduled meeting with any person currently.

> **Example Test Case**: `schedule 1 s/23-04-2024T16:00 e/23-04-2024T17:00`.

Reviewer: ```suggestion
**Example Test Case**: `schedule 1 s/23-04-2025T16:00 e/23-04-2025T17:00`.
```

**Expected Result:**

> 1. A meeting from 23-02-2024 16:00 to 23-02-2024 17:00 is attached to the specified person.

Reviewer: ```suggestion
1. The meeting details of the specified person is updated with the provided rescheduled meeting date and time.
```

- `schedule 5`.
- `schedule 1 s/23-04-2024T00:00 s/23-04-2024T10:00`.
- `schedule 1 s/23-04-2024 e/23-04-2024`.
- `schedule 1 s/31-04-2024T16:00 e/31-04-2024T17:00`.

### Unscheduling a meeting with a person

#### Unscheduling a meeting with a person successfully

**Prerequisites:**
1. List all persons using the `list` command. At least 5 persons are in the list.
2. Assume there is a scheduled meeting with all person currently.

**Example test case:** `unschedule 5`.

**Expected Result:**
1. The meeting detail is deleted for the person at the specified index.
2. *Person List* updates the list of contact with the person with no meeting.
3. *Command Result Box* outputs a successful execution message with the name of the modified person.
4. *Command Box* is cleared.

#### Unschedule for a person without meeting

**Prerequisites:**
1. List all persons using the `list` command. At least 5 persons are in the list.
2. Assume no scheduled meeting with any person currently.

**Example test case:** `unschedule 5`.

**Expected Result:**
1. No person in the contact list is modified.
2. Input in the *Command Box* remains and turns red.
3. Error details are stated in the *Command Result Box*.

> #### Invalid value or command format used for deleting a person

Reviewer 2: I think this is meant to say `unscheduling a meeting with a person` instead of `deleting a person`?

- `remark 100000 r/Birthday on 31 July.`.
- `remark 1 r/Birthday on 31 July r/`.

### Schedule a meeting with a person

#### Successful scheduling

**Prerequisites:**
1. List all persons using the `list` command. At least 5 persons are in the list.

> 2. Assume the time for now is 14-04-2024T12:00.

Author: But for the schedule we need a reference time. I also don't know how can make it more general.

3. Assume no scheduled meeting with any person currently.

**Example Test Case**: `schedule 1 s/23-04-2024T16:00 e/23-04-2024T17:00`.

**Expected Result:**

> 1. A meeting from 23-02-2024 16:00 to 23-02-2024 17:00 is attached to the person at index 1.

Reviewer 3: For the datetime values lets keep it as is since the successful schedule output is as shown:
![image](https://github.com/AY2324S2-CS2103-F15-1/tp/assets/122505234/e6adafd5-657a-414e-8ef0-e92baa03a9f9)

> 1. A meeting from 23-02-2024 16:00 to 23-02-2024 17:00 is attached to the person at index 1.

Author: Yep this one I will change. Thanks!

> 1. A meeting from 23-02-2024 16:00 to 23-02-2024 17:00 is attached to the person at index 1.

Reviewer 2: My main concern right now is due to using assumption, which I feel we should not be doing? If we are not going by that specific assumption the datetime may change accordingly.

**Prerequisites:**
1. List all persons using the `list` command. At least 5 persons are in the list.
2. Assume the time for now is 14-04-2024T12:00.
3. Assume there is a scheduled meeting with all person currently.

**Example Test Case**: `schedule 1 s/23-04-2024T16:00 e/23-04-2024T17:00`.

**Expected Result:**
1. Meeting is **not scheduled** for any person.
2. Input in the *Command Box* remains and turns red.
3. Error details are stated in the *Command Result Box*.

**Alternative Test Cases:**
- `schedule 3 s/23-04-2024T00:00 e/23-04-2025T10:00`.
- `schedule 5 s/23-04-2024T12:00 e/30-04-2024T20:00`.

#### Invalid value or command format used for updating remark of a person

**Prerequisites:**
1. List all persons using the `list` command. Multiple persons are in the list (less than 100000).

**Example Test Case**: `schedule 100001 s/23-04-2024T16:00 e/23-04-2024T17:00`.

**Expected Result:**
1. Meeting is **not scheduled** for any person.
2. Input in the *Command Box* remains and turns red.
3. Error details are stated in the *Command Result Box*.

**Alternative Test Cases:**
- `schedule 1 s/23-04-2024T00:00`.
- `schedule 5`.
- `schedule 1 s/23-04-2024T00:00 s/23-04-2024T10:00`.

> - `schedule 1 s/23-04-2024 s/23-04-2024`.

Reviewer 3: yes we should follow the previous test case conventions. It should also extend to changing 100000 to 100.

> - `schedule 1 s/23-04-2024 s/23-04-2024`.

Author: Wait i see the remark one is still using 100000?

### Unschedule a meeting with a person

#### Unschedule a meeting successfully

**Prerequisites:**
1. List all persons using the `list` command. At least 5 persons are in the list.
2. Assume there is a scheduled meeting with all person currently.

**Example test case:** `unschedule 5`.

**Expected Result:**
1. The meeting detail is deleted for the person at the specified index.
2. *Person List* updates the list of contact with the person with no meeting.
3. *Command Result Box* outputs a successful execution message with the name of the modified person.
4. *Command Box* is cleared.

> 

Reviewer 2: Perhaps you want to add a subsection for cases where there you unschedule a meeting with a person that has a meeting today and state that the meeting list will not include the person once it is unscheduled?

> 

Reviewer 2: Perhaps you want to add a subsection for cases where there you unschedule a meeting with a person that has a meeting today and state that the meeting list will not include the person once it is unscheduled?

- `remark 100000 r/Birthday on 31 July.`.
- `remark 1 r/Birthday on 31 July r/`.

### Scheduling a meeting with a person

#### Scheduling a meeting with a person successfully

> ##### The time input of this test must be after current date time.

Reviewer 2: I think this should not be a header? I think we should use a text either in a `<div>` block or simply a bolded text?

> ##### The time input of this test must be after current date time.

Reviewer: ```suggestion
##### The provided datetimes must be after system's current datetime.
```

**Prerequisites:**
1. List all persons using the `list` command. At least 5 persons are in the list.
2. No person in the list has any scheduled meeting.

> **Example Test Case**: `schedule 1 s/23-04-2025T16:00 e/23-04-2025T17:00`.

Reviewer 2: I think the note that the input start datetime and end datetime should be placed here?

**Expected Result:**
1. The meeting details of the specified person is updated with the provided meeting date and time.
2. *Person List* updates the list of contact with the person with details of the meeting.
3. *Command Result Box* outputs a successful execution message with the details of the meeting with the specified person.
4. *Command Box* is cleared.

> 

Reviewer 2: I think we are missing alternative test cases that adds a meeting remark?

#### Scheduling a meeting today successfully

> ##### The time of this test must be before 23:00 of the current date

Reviewer 2: This should not be a header? I think this should fall under a prerequisite?

**Example Test Case**: `schedule 100001 s/23-04-2024T16:00 e/23-04-2024T17:00`.

**Expected Result:**
1. Meeting is **not scheduled** for any person.
2. Input in the *Command Box* remains and turns red.
3. Error details are stated in the *Command Result Box*.

**Alternative Test Cases:**
- `schedule 1 s/23-04-2024T00:00`.
- `schedule 5`.
- `schedule 1 s/23-04-2024T00:00 s/23-04-2024T10:00`.
- `schedule 1 s/23-04-2024 e/23-04-2024`.
- `schedule 1 s/31-04-2024T16:00 e/31-04-2024T17:00`.

### Unscheduling a meeting with a person

#### Unscheduling a meeting with a person successfully

**Prerequisites:**
1. List all persons using the `list` command. At least 5 persons are in the list.
2. All person in the list has a scheduled meeting.

**Example test case:** `unschedule 5`.

**Expected Result:**
1. The meeting detail is deleted for the specified person.
2. *Person List* updates the list of contact with the person with no meeting.
3. *Command Result Box* outputs a successful execution message with the name of the specified person.
4. *Command Box* is cleared.

#### Unscheduling a meeting today successfully

> ##### The time of this test must be before 23:00 of the current date

Reviewer 2: This should not be a header? I think this should fall under a prerequisite?

> ##### The time of this test must be before 23:00 of the current date

Reviewer: ```suggestion
##### The time of this test must be before 23:00 of system's current date
```
